### PR TITLE
Substrait Interface in R API

### PR DESCRIPTION
**Pull request by pdet**
Implements the following functions for substrait in the R API
```r
 plan <- duckdb::duckdb_get_substrait(con, query) #Serializes query into substrait plan
 result <- duckdb::duckdb_prepare_substrait(con, plan) #Executes substrait plan 
```

cc: @jonkeane

### CODE REVIEW

File: tools/rpkg/tests/testthat/_snaps/explain.md [hunk 1]
```diff
@@ -1,71 +0,0 @@
-# EXPLAIN gives reasonable output
```
Comment (rsund) [inline tools/rpkg/tests/testthat/_snaps/explain.md:1]: It seems that this file required for snapshot tests was accidentally(?!) deleted. If there is no particular reason to delete, please don't.

File: tools/rpkg/tests/testthat/_snaps/multi_statement.md [hunk 2]
```diff
@@ -1,31 +0,0 @@
-# empty statement gives an error
```
Comment (rsund) [inline tools/rpkg/tests/testthat/_snaps/multi_statement.md:1]: Here is another file that probably should not be deleted.

File: tools/rpkg/R/substrait.R [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#' Query DuckDB using Substrait
+#' TODO document this
```
Comment (hannes) [inline tools/rpkg/R/substrait.R:2]: Maybe add a sentence here and below?

File: tools/rpkg/src/cpp11.cpp [hunk 4]
```diff
@@ -267,6 +281,8 @@ static const R_CallMethodDef CallEntries[] = {
     {"_duckdb_rapi_execute",          (DL_FUNC) &_duckdb_rapi_execute,          2},
     {"_duckdb_rapi_execute_arrow",    (DL_FUNC) &_duckdb_rapi_execute_arrow,    2},
     {"_duckdb_rapi_expr_constant",    (DL_FUNC) &_duckdb_rapi_expr_constant,    1},
+ 	{"_duckdb_rapi_get_substrait",     (DL_FUNC) &_duckdb_rapi_get_substrait,     2},
+{"_duckdb_rapi_prepare_substrait", (DL_FUNC) &_duckdb_rapi_prepare_substrait, 2},
```
Comment (hannes) [inline tools/rpkg/src/cpp11.cpp:285]: Indentation seems off here

File: tools/rpkg/tests/testthat/_snaps/explain.md (hunk 1 above)
Comment (pdet) [reply tools/rpkg/tests/testthat/_snaps/explain.md:1]: Yeah, I'm not sure how these got deleted, thanks for spotting it.

**Pull request merged**